docs/src/index.md

```diff
@@ -82,3 +82,59 @@ The laziness of the main interfaces are summarized below:
+will match the corresponding branches:
+
+```julia-repl
+julia> f = UnROOT.samplefile("km3net_offline.root")
+ROOTFile with 2 entries and 25 streamers.
+/Users/tamasgal/Dev/UnROOT.jl/test/samples/km3net_offline.root
+├─ E (TTree)
+│  └─ "Evt"
+└─ Header (Head)
+
+
+julia> t = LazyTree(f, "E", ["Evt/trks/trks.id", r"Evt/trks/trks.pos.[xyz]"])
+ Row │ Evt_trkstrksid                Evt_trkstrkspos               Evt_trkstrkspos               Evt_trkstrkspos ⋯     │ SubArray{Int32,               SubArray{Float6               SubArray{Float6               SubArray{Float6 ⋯─────┼──────────────────────────────────────────────────────────────────────────────────────────────────────────── 1   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [125.0, 125.0, 70.7, 67.2, 6  [615.0, 615.0, 585.0, 583.0,  [446.0, 446.0,  ⋯ 2   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [80.7, 80.7, 39.1, 33.2, 51.  [533.0, 533.0, 559.0, 560.0,  [465.0, 465.0,  ⋯ 3   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [194.0, 194.0, 96.5, 92.5, 9  [593.0, 593.0, 581.0, 580.0,  [457.0, 457.0,  ⋯ 4   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [204.0, 204.0, 124.0, 134.0,  [590.0, 590.0, 571.0, 574.0,  [440.0, 440.0,  ⋯ 5   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [58.6, 58.6, 30.1, 35.7, 34.  [546.0, 546.0, 565.0, 562.0,  [440.0, 440.0,  ⋯ 6   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [202.0, 202.0, 183.0, 185.0,  [585.0, 585.0, 564.0, 579.0,  [424.0, 424.0,  ⋯ 7   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [47.3, 47.3, 30.1, 24.3, 26.  [533.0, 533.0, 557.0, 554.0,  [440.0, 440.0,  ⋯ 8   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [200.0, 200.0, 179.0, 180.0,  [569.0, 569.0, 555.0, 575.0,  [469.0, 469.0,  ⋯ 9   │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [209.0, 209.0, 101.0, 102.0,  [557.0, 557.0, 575.0, 580.0,  [412.0, 412.0,  ⋯ 10  │ [1, 2, 3, 4, 5, 6, 7, 8, 9,   [172.0, 172.0, 126.0, 125.0,  [532.0, 532.0, 572.0, 572.0,  [443.0, 443.0,  ⋯                                                                                                  1 column omitted
+
+julia> names(t)
+4-element Vector{String}:
+ "Evt_trkstrksid"
+ "Evt_trkstrksposz"
+ "Evt_trkstrksposy"
+ "Evt_trkstrksposx"
+ 
+julia> t.Evt_trkstrksposy
+10-element LazyBranch{SubArray{Float64, 1, Vector{Float64}, Tuple{UnitRange{Int64}}, true}, UnROOT.Nooffsetjagg, ArraysOfArrays.VectorOfVectors{Float64, Vector{Float64}, Vector{Int32}, Vector{Tuple{}}}}: 
+ [615.1089636184813, 615.1089636184813, 584.7490001284564, 582.9922451367319, 583.4532742304276, 583.212063675951, 583.182239372315, 582.5351568422853, 581.7743452806689, 583.6562661040083  …  597.4671030654561, 575.2131287145014, 582.7727094920472, 588.1060616488143, 574.8482883396676, 603.955888460846, 593.4457411811859, 576.6419859859786, 574.836340445788, 576.5382993955498]
+ ...
+ ...
+```
+
+Branch names are normalised so that they contain valid characters for
+identifiers. The branchname `Evt/trks/trks.pos.y` for example is therefore
+converted to `Evt_trkstrksposy`, which might be a bit inconvenient to use.
```

Comment:
this looks weird, I thought we would have `/ -> _` for each occurrence not just the first one, maybe it's a long standing bug

Reply:
Ah OK, I thought that's intentional. Should we fix it here? It will be a breaking change though...

Reply:
we're going 0.9 -> 0.10 anyway xd

Reply:
True ;)

Reply:
Not sure about the fix. I just realise that it always replaces all occurrences of `head`, which might be not what we want (actually it could break the naming). We should probably only remove consecutive occurences of the same thing.

Reply:
OK, fixed